src/devices/src/virtio/block/request.rs

```diff
@@ -424,152 +424,213 @@ mod tests {
         }
     }
 
-    #[test]
-    #[allow(clippy::cognitive_complexity)]
-    fn test_parse() {
-        let m = &GuestMemoryMmap::from_ranges(&[(GuestAddress(0), 0x10000)]).unwrap();
-        let vq = VirtQueue::new(GuestAddress(0), &m, 16);
+    struct RequestVirtQueue<'a> {
```

Comment:
Nice abstraction! The naming is not expressive enough, though I couldn't find a better one.

Reply:
I can't think of a better name now. Anyway we can change it in the future.